src/pages/en/core-concepts/framework-components.md

```diff
@@ -0,0 +1,143 @@
+---
+layout: ~/layouts/MainLayout.astro
+title: Framework Components
+description: Learn how to use React, Svelte, etc.
+---
+Astro allows you to build your site using JavaScript UI components from a variety of popular frameworks. Astro will run your [React](https://reactjs.org/), [Preact](https://preactjs.com/), [Svelte](https://svelte.dev/), [Vue](https://vuejs.org/), [SolidJS](https://www.solidjs.com/), [AlpineJS](https://alpinejs.dev/) and [Lit](https://lit.dev/) components at build time, then render your entire site to static HTML by default, removing all the JavaScript from the final page!
+
+You can use any of these frameworks as templating languages to build static pages, taking advantage of the syntax and tooling you already know and love. This allows you to build with components you're familiar with, maybe even the exact same ones you've already written! And, you won't send any unnecessary JavaScript to the client to render static elements, even dynamically generated ones.
+
+For client-side interactivity, Astro allows you to opt-in to load a single component's JavaScript and dpendencies only when and where that component is used.
+
+> 📚 Read more about Astro's [partial hydration](en/core-concepts/partial-hydration)
+
+## Configure Framework Renderers
+
+Your Astro configuration must include the renderer for each framework you use. You can find, and manually edit, your project's current list of available renderers in `astro.config.mjs`.
+
+```js
+export default ({
+// No renderers are needed for AlpineJS support, just use Astro components!
+renderers: [
+    '@astrojs/renderer-svelte',
+    '@astrojs/renderer-vue',
+    '@astrojs/renderer-react',
+    '@astrojs/renderer-preact',
+    '@astrojs/renderer-lit',
+    '@astrojs/renderer-solid',
+  ],
+
+});
+```
+
+⚙️ See the [Renderer Reference](/en/reference/renderer-reference) to learn about configuring custom renderers.
+
+## Build With Components
+
+Use your JavaScript framework components in your Astro pages, layouts and components just like Astro components! All your components can live together in `/src/components`, or can be organized in any way you like.
+
+### Render on a Page
+
+To use a framework component, import it from its relative path, including file extension, in the component script. Then, use the component alongside other components, HTML elements and JSX-like expressions in the component template. 
+```astro
+---
+import MyReactComponent from '../components/MyReactComponent.jsx';
+import MySvelteComponent from '../components/MySvelteComponent.svelte';
+import MyAstroComponent from '../components/MyAstroComponent.astro';
+const title="My Astro Page"
+---
+<h1>{title}</h1>
+<My AstroComponent />
+
+<h2>Using a Svelte Component:</h2>
+<MySvelteComponent client: load>
+    <p>Here is some text.<p>
+</MySvelteComponent>
+
+<h2>Using a React Component:</h2>
+<MyReactComponent client:visible />
```

Comment:
So, I was torn on this and I'm open to suggestion. Since we haven't really discussed hydration yet, I know it's a little risky to "clutter" the example with the client directives here.

BUT, since SO much of the time people's initial problems are that they DON'T use a client directive (until they get the hang of it), I'm hesitant to show them of an example WITHOUT a client directive, lest they internalize it. 

Counter-counter-point: If we've made a big deal about how they can get zero JS, then maybe we should at least show an example that achieves that?

Reply:
I made the call to go with your suggestion, @delucis. I think it's worth showing a non-hydrated component in this example. Thanks for the improvement!